src/dbnode/storage/shard.go

```diff
@@ -2144,18 +2157,36 @@ func (s *dbShard) removeAnyFlushStatesTooEarly(tickStart time.Time) {
 
 func (s *dbShard) CleanupExpiredFileSets(earliestToRetain time.Time) error {
 	filePathPrefix := s.opts.CommitLogOptions().FilesystemOptions().FilePathPrefix()
-	multiErr := xerrors.NewMultiError()
-	expired, err := s.filesetBeforeFn(filePathPrefix, s.namespace.ID(), s.ID(), earliestToRetain)
+	expired, err := s.filesetPathsBeforeFn(filePathPrefix, s.namespace.ID(), s.ID(), earliestToRetain)
 	if err != nil {
-		detailedErr :=
-			fmt.Errorf("encountered errors when getting fileset files for prefix %s namespace %s shard %d: %v",
-				filePathPrefix, s.namespace.ID(), s.ID(), err)
-		multiErr = multiErr.Add(detailedErr)
+		return fmt.Errorf("encountered errors when getting fileset files for prefix %s namespace %s shard %d: %v",
+			filePathPrefix, s.namespace.ID(), s.ID(), err)
 	}
-	if err := s.deleteFilesFn(expired); err != nil {
-		multiErr = multiErr.Add(err)
+
+	return s.deleteFilesFn(expired)
+}
+
+func (s *dbShard) CleanupCompactedFileSets() error {
```

Comment:
like I was saying before, probably want this in `cleanup.go`. Not necessary in this P.R if we dont want to

Reply:
Will track this in https://github.com/m3db/m3/issues/1748.